doc/arch/adr-002-rest-api.md

```diff
@@ -0,0 +1,74 @@
+| `/boundaries/{id}/reference-images/`       | POST   | `{ ... }`                | Contributor    | 200 OK,<br />404 Not Found                                 | Adds a new reference image to the boundary                                                                         |
+| `/boundaries/{id}/submit/`                 | POST   | `{ notes }`              | Contributor    | 200 OK,<br />404 Not Found                                 | Submits the boundary                                                                                               |
+| `/boundaries/{id}/review/`                 | POST   | `{ notes, annotations }` | Validator      | 200 OK,<br />404 Not Found                                 | Reviews a boundary                                                                                                 |
+| `/boundaries/{id}/review/annotations/{id}` | PUT    | `{ annotation }`         | Validator      | 200 OK,<br />404 Not Found                                 | Updated an annotation                                                                                              |
+| `/boundaries/{id}/approve/`                | POST   | -                        | Validator      | 200 OK,<br />404 Not Found                                 | Approves a boundary                                                                                                |
+| `/boundaries/{id}/unapprove/`              | POST   | -                        | Validator      | 200 OK,<br />404 Not Found                                 | Unapproves a boundary                                                                                              |
+
+### User
+
+| Path                 | Method | Body      | Authentication | Responses                  | Notes                              |
+| -------------------- | ------ | --------- | -------------- | -------------------------- | ---------------------------------- |
+| `/user/{id}/profile` | PUT    | `{ ... }` | Contributor    | 200 OK,<br />404 Not Found | Update contributor contact details |
+
+## Notes
+
+Any thing that a Contributor or Validator can do, an Administrator can also do.
+
+A payload Body of `-` means that there is no body required. A payload Body of `{ ... }` means that there is a required body, but is not elaborated here.
+
+All endpoints that are marked as requiring a Logged In User will also return 401 Not Authorized if accessed without proper credentials.
+
+When we return `404 Not Found` for objects that the User does not have permissions to, we're using a pattern like this:
+
+```python
+@permission_classes((IsAuthenticated,))
+def get_boundary(request, id):
+    user = request.user
+    boundary = get_object_or_404(Boundary, id)
+    submission = boundary.get_latest_submission_for(user)
+
+    serializer = SubmissionSerializer(submission)
```

Comment:
This is also confusing, because the endpoint is called `get_boundary` but we're returning a serialized `submission`.

Perhaps there can be a `BoundaryDetailSerializer` that takes a `boundary` and does the `get_latest_submission_for(user)` within it, so we can return that here.